### DISCUSSION TITLE
Route Output to Input

### DISCUSSION THREAD

**Discussion by Stektax** [MusicPlayerDaemon/MPD#2110] [Q&A] [answered]
Hello, i am trying to route the output of MPD to a (virtual) input device.
The system should be able to pick up the output of MPD as input.

More specifically the project [rAudio](https://github.com/rern/rAudio) uses MPD and i want to route it there as said above.

Does anyone know a way? Until now i only found ways to route input to output, not the other way around.

**Comment by borine** [comment]
> The system should be able to pick up the output of MPD as input.

What do you mean by "the system" ? If you mean an ALSA input device, then I guess you should try the ALSA Loopback card (snd-aloop kernel module). You can load this driver module with:
```
sudo modprobe snd-aloop
```
Then add an audio_output entry to your mpd.conf for a device on this card; for example
```
audio_output {
    type "alsa"
    name "Loopback"
    device "hw:Loopback,0"
    mixer_type "software"
}
```
Then any application can capture from `hw:Loopback,1` to pick up the mpd output. You would need to search general linux audio forums for help with `snd-aloop` if you need it.

**Comment by MaxKellermann** [comment, marked as answer]
The ALSA loopback driver was designed for this. It's not MPD specific - any program using ALSA can do this.
